app/(app)/offence/index.tsx

```diff
@@ -1,26 +1,31 @@
 import { Link, router } from 'expo-router'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import { ScrollView } from 'react-native'
 
 import SelectButton from '@/components/inputs/SelectButton'
 import TextInput from '@/components/inputs/TextInput'
 import SelectRow from '@/components/list-rows/SelectRow'
+import LocationMapPreview from '@/components/map/location-map/LocationMapPreview'
 import ContinueButton from '@/components/navigation/ContinueButton'
 import ScreenContent from '@/components/screen-layout/ScreenContent'
 import ScreenView from '@/components/screen-layout/ScreenView'
 import Field from '@/components/shared/Field'
+import PressableStyled from '@/components/shared/PressableStyled'
 import { getOffenceTypeLabel } from '@/modules/backend/constants/offenceTypes'
 import { getResolutionTypeLabel } from '@/modules/backend/constants/resolutionTypes'
 import { getRoleByKey } from '@/modules/backend/constants/roles'
+import { useLocation } from '@/modules/map/hooks/useLocation'
 import { useOffenceStoreContext } from '@/state/OffenceStore/useOffenceStoreContext'
 import { useSetOffenceState } from '@/state/OffenceStore/useSetOffenceState'
 
 // TODO - move to translations JSON after this feature is added
 const requiredText = 'Toto pole je povinné'
 
 const OffencePage = () => {
-  const { ecv, offenceType, roleKey, resolutionType, isObjectiveResponsibility } =
+  const { ecv, offenceType, roleKey, resolutionType, isObjectiveResponsibility, location } =
     useOffenceStoreContext((state) => state)
   const setState = useSetOffenceState()
```

Comment:
--> `setOffenceState`

In my opinion, it would be better if the `useSetOffenceState` returns the object with the function, so it's used always with the same name.